File: task_equivalents/llm_translations/gpt/base/r/machine_learning_workflows/02_regression_workflow_feature_importance.R
library(readr)
library(dplyr)
library(glmnet)
library(gbm)

Hitters <- read_csv("data/csv/Hitters.csv") %>%
na.omit()

cols_to_drop <- intersect(c("rownames", "Player", "Name"), names(Hitters))
Hitters <- Hitters %>%
select(-all_of(cols_to_drop))

Hitters <- Hitters %>%
mutate(
League = recode(League, "A" = 0, "N" = 1),
NewLeague = recode(NewLeague, "A" = 0, "N" = 1),
Division = recode(Division, "E" = 0, "W" = 1),
logSal = log(Salary)
)

x <- Hitters %>%
select(-Salary, -logSal)
y <- Hitters$logSal

set.seed(1)
train_idx <- sample(seq_len(nrow(x)), 200)
xtrain <- x[train_idx, ]
xtest <- x[-train_idx, ]
ytrain <- y[train_idx]
ytest <- y[-train_idx]

lambda_grid <- seq(0.001, 0.201, length.out = 20)
train_mse <- numeric(20)

for (i in seq_along(lambda_grid)) {
boost <- gbm(
formula = ytrain ~ .,
distribution = "gaussian",
data = data.frame(ytrain = ytrain, xtrain),
n.trees = 1000,
interaction.depth = 1,
shrinkage = lambda_grid[i],
bag.fraction = 0.5,
train.fraction = 1.0,
verbose = FALSE
)

pred_train <- predict(boost, newdata = xtrain, n.trees = 1000)
train_mse[i] <- mean((ytrain - pred_train)^2)
}

xtrain_mat <- model.matrix(~ ., data = xtrain)[, -1]
xtest_mat <- model.matrix(~ ., data = xtest)[, -1]

lasso <- cv.glmnet(xtrain_mat, ytrain, alpha = 1, nfolds = 10)
lasso_pred <- predict(lasso, s = "lambda.min", newx = xtest_mat)
lasso_mse <- mean((ytest - lasso_pred)^2)

best_lambda <- lambda_grid[which.min(train_mse)]
best_boost <- gbm(
formula = ytrain ~ .,
distribution = "gaussian",
data = data.frame(ytrain = ytrain, xtrain),
n.trees = 1000,
interaction.depth = 1,
shrinkage = best_lambda,
bag.fraction = 0.5,
train.fraction = 1.0,
verbose = FALSE
)

feature_importance <- summary(best_boost, plotit = FALSE)
feature_importance <- feature_importance[order(-feature_importance$rel.inf), ]
print(setNames(feature_importance$rel.inf, feature_importance$var))

File: task_equivalents/llm_translations/gpt/base/python/machine_learning_workflows/02_regression_workflow_feature_importance.py
import numpy as np
import pandas as pd
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.linear_model import LassoCV

Hitters = pd.read_csv('data/csv/Hitters.csv', header=0, index_col=0)

Hitters = Hitters.dropna()
drop_cols = [col for col in ['rownames', 'Player', 'Name'] if col in Hitters.columns]
Hitters = Hitters.drop(columns=drop_cols)
Hitters['League'] = np.where(Hitters['League'] == 'A', 0, 1)
Hitters['NewLeague'] = np.where(Hitters['NewLeague'] == 'A', 0, 1)
Hitters['Division'] = np.where(Hitters['Division'] == 'E', 0, 1)
Hitters['logSal'] = np.log(Hitters['Salary'])

x = Hitters.drop(columns=['Salary', 'logSal'])
y = Hitters['logSal']

np.random.seed(1)
train = np.random.choice(np.arange(len(x)), size=200, replace=False)
test = np.setdiff1d(np.arange(len(x)), train)
xtrain = x.iloc[train]
xtest = x.iloc[test]
ytrain = y.iloc[train]
ytest = y.iloc[test]

train_mse = np.zeros(20)
lambda_grid = np.linspace(0.001, 0.201, 20)

for i, lam in enumerate(lambda_grid):
    boost_hit = GradientBoostingRegressor(
        n_estimators=1000,
        learning_rate=lam,
        random_state=1
    )
    boost_hit.fit(xtrain, ytrain)
    boost_pred = boost_hit.predict(xtrain)
    train_mse[i] = np.mean((boost_pred - ytrain) ** 2)

cv_out = LassoCV(cv=10, random_state=1)
cv_out.fit(xtrain.to_numpy(), ytrain.to_numpy())
bestlam = cv_out.alpha_
lasso_pred = cv_out.predict(xtest.to_numpy())
lasso_mse = np.mean((lasso_pred - ytest.to_numpy()) ** 2)

best_lambda = lambda_grid[np.argmin(train_mse)]
best_boost_hit = GradientBoostingRegressor(
    n_estimators=1000,
    learning_rate=best_lambda,
    random_state=1
)
best_boost_hit.fit(xtrain, ytrain)

feature_importance = pd.Series(
    best_boost_hit.feature_importances_,
    index=xtrain.columns
).sort_values(ascending=False)
print(feature_importance)